### PR TITLE
Fix navigation editor link search suggestions

### PR DESCRIPTION
**Pull request by talldan**
## Description
#29012 caused a regression in the link suggestions for the navigation editor, resulting in some very long failing HTTP requests:
Screenshot 2021-03-10 at 5 41 13 pm

The line of code that cause the issue is:
https://github.com/WordPress/gutenberg/pull/29012/files#diff-8b871841ffe5cb86e63dd3e6df4e2230028ca02ff8ecebad37fac39cf5aa2b99R29-R30

This provided no way for the search text or search options to be passed into the `fetchLinkSuggestions` function. It's remedied in this PR.

## How has this been tested?
1. Open the navigation editor.
2. Add a link to a menu
3. Search for an existing page/post (this won't work if you have no pages or posts)
4. Search results should display

## Screenshots 
Screenshot 2021-03-10 at 5 39 46 pm

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by draganescu — APPROVED** [review]
Tested and fixes the regression. Great that we have a test for this now :)

File: packages/edit-navigation/src/components/editor/style.scss [hunk 1]
```diff
@@ -103,7 +103,6 @@
 
 	// Add Buttons
 	.block-editor-button-block-appender.block-list-appender__toggle {
-		margin: 0 0 0 $grid-unit-20;
```
Comment (gwwar) [inline packages/edit-navigation/src/components/editor/style.scss:106]: Was this change intentional?

File: packages/e2e-tests/specs/experiments/navigation-editor.test.js [hunk 2]
```diff
@@ -282,4 +328,60 @@ describe( 'Navigation editor', () => {
 		} );
 		expect( submenuLinkHidden ).toBeDefined();
 	} );
+
+	it( 'displays suggestions when adding a link', async () => {
```
Comment (gwwar) [inline packages/e2e-tests/specs/experiments/navigation-editor.test.js:332]: 👏 Thanks for adding an additional e2e test here

**Review by gwwar — APPROVED** [review]
Thanks @talldan this tests well for me! This is good to land once the E2Es are fixed (or put into a follow up PR).

https://user-images.githubusercontent.com/1270189/110711030-b6be8380-81b3-11eb-8a79-b86d4950caa6.mp4

Running `npm run test-e2e -- --puppeteer-interactive packages/e2e-tests/specs/experiments/navigation-editor.test.js` locally I see

```
  ● Navigation editor › displays suggestions when adding a link

    TimeoutError: waiting for selector `button[aria-label="Add Link"]` failed: timeout 30000ms exceeded

      349 | 
      350 | 		// Add an inner link block.
    > 351 | 		const appender = await page.waitForSelector(
          | 		                            ^
      352 | 			'button[aria-label="Add Link"]'
      353 | 		);
      354 | 		await appender.click();
```

**Comment by talldan** [comment]
@gwwar Thanks for reviewing. Yep, I'm having trouble getting the new end to end test to pass. When running on CI, the Navigation Link block seems to have no variations. Clicking the appender inserts the block directly rather than opening the quick inserter.

If you revert the changes in 88068a7 it should run well locally, but fails in CI.

Slightly baffled by this 🤔

File: packages/edit-navigation/src/components/editor/style.scss (hunk 1 above)
Comment (talldan) [reply packages/edit-navigation/src/components/editor/style.scss]: I thought it might be related to the test failure, but nope. I've dropped the commit from this branch.

**Comment by talldan** [comment]
I've updated the test to work for no variations, but this means it won't pass locally. Would be good to understand what's happening here.

**Comment by gwwar** [comment]
> When running on CI, the Navigation Link block seems to have no variations.

Curious 🤔  @talldan one thing I changed recently was moving the fallback hardcoded variations to the `blocks.registerBlockType` filter. See https://github.com/WordPress/gutenberg/pull/29095 for more context.  It could be that the e2e setup is doing something funny there.

https://github.com/WordPress/gutenberg/blob/647a41cbab8e44e5ecd79af381f54cec080a3758/packages/block-library/src/navigation-link/hooks.js#L71-L75

**Comment by talldan** [comment]
The bit that's unusual is that it works fine locally, the setup should be exactly the same.

I thought it might be that one of the test plugins was somehow staying active, but I've tested those locally and couldn't see any issues. But it could still be another test leaving behind some stray configuration.
